Dockerfile.dapper

```diff
@@ -9,7 +9,7 @@ ENV CATTLE_K3S_VERSION v1.21.7+k3s1
 ENV CATTLE_HELM_UNITTEST_VERSION v0.1.7-rancher3
 # helm 3 version
 ENV HELM_VERSION v3.7.1
-ENV KUSTOMIZE_VERSION v3.10.0
+ENV KUSTOMIZE_VERSION v4.4.1
```

Comment:
I can't find anything deprecated in the kustomize v4.x release notes that I can think it would break anything. But I don't understand how kustomize is used in Rancher, so I don't know how to test it. 
All calls to kustomize.sh I find are in `managementuserlegacy`  For example https://github.com/rancher/rancher/blob/725a0b4702670616bc5dff507bd9fec0720676c0/pkg/controllers/managementuserlegacy/helm/common/common.go#L348 What does `legacy` mean here?

@luthermonson could you please explain me how kustomize is used? so I can test v.4.4.1 to make sure we are not breaking anything.

Reply:
i dont have that much of an understanding. i tagged two people in your slack convo to get it going more. my only understanding is we are wrapping some helm stuff with kustomize build commands so it's probably all charts/apps&marketplace code so i tagged two people in that space to take a look. my gut tells me we'll be fine cause i dont think `kustomize build` has changed much but you never know so better safe than sorry before we commit to a major bump in the executable

Reply:
We used it before as a post renderer to add some MatchLabel Selectors. We then found that because of this there were compatibility issues when upgrading an old app to a new app, so we also have added it to the [shell image](https://github.com/rancher/shell). We may want to bump kustomize in the shell image as well to make sure they are compatible.

Reply:
@rmweir kustomize in shell is already bumped to v4.4.1 and merged into master https://github.com/rancher/shell/blob/master/package/Dockerfile#10

Reply:
I did a manual test, and kustomize v4.4.1 works fine.